app/src/main/java/nodomain/freeyourgadget/gadgetbridge/adapter/GBDeviceAdapterv2.java

```diff
@@ -290,6 +303,100 @@ public void onDismissed(Snackbar snackbar, int event) {
 
         );
 
+        holder.fmFrequencyBox.setVisibility(View.GONE);
+        if (device.isInitialized() && device.getFmFrequency() != -1) {
+            holder.fmFrequencyBox.setVisibility(View.VISIBLE);
+            holder.fmFrequencyLabel.setText(String.format(Locale.getDefault(), "%.1f", device.getFmFrequency()));
+        }
+        final TextView fmFrequencyLabel = holder.fmFrequencyLabel;
+        holder.fmFrequencyBox.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                AlertDialog.Builder builder = new AlertDialog.Builder(context);
+                builder.setTitle(R.string.preferences_fm_frequency);
+
+                final EditText input = new EditText(context);
+
+                input.setSelection(input.getText().length());
+                input.setRawInputType(InputType.TYPE_CLASS_NUMBER | InputType.TYPE_NUMBER_FLAG_DECIMAL);
+                input.setText(String.format(Locale.getDefault(), "%.1f", device.getFmFrequency()));
+                builder.setView(input);
+
+                builder.setPositiveButton(context.getResources().getString(android.R.string.ok),
+                        new DialogInterface.OnClickListener() {
+                            @Override
+                            public void onClick(DialogInterface dialog, int which) {
+                                float frequency = Float.valueOf(input.getText().toString());
+                                frequency = Float.valueOf(String.format(Locale.getDefault(), "%.1f", frequency));
```

Comment:
Just to limit it to 1 decimal place, since it doesn't make sense to have more. Maybe there was a better way to do it, and maybe the user should be warned?

Reply:
Ah, I see. If you could make that a separate method with a speaking name or add a comment about this, that would be sufficient, IMHO.